samcli/local/apigw/path_converter.py

```diff
@@ -4,16 +4,31 @@
 
 import re
 
-FLASK_PATH_PARAMS = "/<path:proxy>"
-APIGW_PATH_PARAMS_ESCAPED = r"/{proxy\+}"
-APIGW_PATH_PARAMS = "/{proxy+}"
+# The regex captures any path information before the {proxy+}. This is to support paths that have other params in
+# them. Example: /id/{id}/user/{proxy+}. Otherwise the regex will match the first { with the last +} giving incorrect
+# results.
+APIGW_PATH_PARAMS_ESCAPED = r"(.*/){(.*)\+}"
+
+# The regex replaces what was captured with APIGW_PATH_PARAMS_ESCAPED to construct the full path with the {proxy+}
+# replaces. The first group is anything before {proxy+}, while the second group is the name given to the proxy.
+# Example: /id/{id}/user/{resource+}; g<1> = '/id/{id}/user/'; g<2> = 'resource'
+FLASK_PATH_PARAMS = r"\g<1><path:\g<2>>"
```

Comment:
nit: `PATH_PARAMS` suffix isn't too clear. Consider dropping the suffix or changing to something else

Reply:
Updated to: `FLASK_CAPTURE_ALL_PATH`